scheduler/client/cli/sched_alg_params.go

```diff
@@ -0,0 +1,170 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type lbsSchedAlgParams struct {
+	ClassLoadPcts        map[string]int32
+	RequestorMap         map[string]string
+	RebalanceMinDuration int
+	RebalanceThreshold   int
+
+	clpFilePath    string
+	reqMapFilePath string
+}
+type getLBSSchedAlgParams struct {
+	printAsJSON bool
+	params      lbsSchedAlgParams
+}
+
+func (g *getLBSSchedAlgParams) registerFlags() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "get_scheduling_alg_params",
+		Short: "GetSchedAlgParams",
+	}
+	r.Flags().BoolVar(&g.printAsJSON, "json", false, "Print out scheduling algorithm parameters as JSON")
+	return r
+}
+
+func (g *getLBSSchedAlgParams) run(cl *simpleCLIClient, cmd *cobra.Command, args []string) error {
+	log.Info("Getting Scheduling Algorithm Parameters", args)
+
+	var err error
+	g.params.ClassLoadPcts, err = cl.scootClient.GetClassLoadPcts()
+	if err != nil {
+		return getReturnError(err)
+	}
+
+	g.params.RequestorMap, err = cl.scootClient.GetRequestorToClassMap()
+	if err != nil {
+		return getReturnError(err)
+	}
+
+	var tInt int32
+	tInt, err = cl.scootClient.GetRebalanceMinDuration()
+	if err != nil {
+		return getReturnError(err)
+	}
+	g.params.RebalanceMinDuration = int(tInt)
+
+	tInt, err = cl.scootClient.GetRebalanceThreshold()
+	if err != nil {
+		return getReturnError(err)
+	}
+	g.params.RebalanceThreshold = int(tInt)
+
+	if g.printAsJSON {
+		asJSON, err := json.Marshal(g.params)
+		if err != nil {
+			log.Errorf("Error converting status to JSON: %v", err.Error())
+			return fmt.Errorf("Error converting status to JSON: %v", err.Error())
+		}
+		log.Infof("%s\n", string(asJSON))
+		fmt.Printf("%s\n", string(asJSON)) // must also go to stdout in case caller looking in stdout for the results
+	} else {
+		log.Info("Class Load Percents:")
+		fmt.Println("Class Load Percents:")
+		for class, pct := range g.params.ClassLoadPcts {
+			log.Infof("%s:%d", class, pct)
+			fmt.Println(class, ":", pct)
+		}
+		log.Info("Requestor (reg exp) to class map:")
+		fmt.Println("Requestor (reg exp) to class map:")
+		for requestorRe, class := range g.params.RequestorMap {
+			log.Infof("%s:%s", requestorRe, class)
+			fmt.Println(requestorRe, ":", class)
+		}
+		log.Infof("Rebalance Duration:%d (minutes)", g.params.RebalanceMinDuration)
+		fmt.Println("Rebalance Duration:", g.params.RebalanceMinDuration, " (minutes)")
+		log.Infof("Rebalance Threshold:%d", g.params.RebalanceThreshold)
+		fmt.Println("Rebalance Threshold:", g.params.RebalanceThreshold)
+
+	}
+
+	return nil
+}
+
+func (s *lbsSchedAlgParams) registerFlags() *cobra.Command {
```

Comment:
IMO, modules shouldn't have 2 registerFlags functions. Could you explain the need for two?

Reply:
we're starting to get a lot of subcommands (with 10 files for handling them).  It seems natural to me to group the set/get pairs together rather than have them in 2 different files.